spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/MigrateTableSparkAction.java

```diff
@@ -56,17 +56,16 @@ public class MigrateTableSparkAction extends BaseTableCreationSparkAction<Migrat
 
   private final StagingTableCatalog destCatalog;
   private final Identifier destTableIdent;
-  private final Identifier backupIdent;
 
   private boolean dropBackup = false;
 
+  private String backupTableName = "";
```

Comment:
What about placing this var next to `dropBackup`? Both of them are non-final variables that can be overridden.

Also, what about just making `backupIdent` non-final but keeping the initialization in the constructor? We can call construct an identifier in `withBackupTableName`. That way, we should be able to reduce the amount of changes.

Reply:
Thanks for the suggestion. The suggestion totally makes sense to me. Based on this comment, I update as follows:
* revert back the `backupIdent` variable with making the var *non-final*, and the `backupIdent` initialization part
* add the table change logic in the `withBackupTableName` method that is newly added in this PR
* remove the `backupTableName` var along with the way to update back along with the above two changes